test/e2e/network/netpol/probe.go

```diff
@@ -83,11 +89,24 @@ func ProbePodToPodConnectivity(k8s *kubeManager, model *Model, testCase *TestCas
 
 // probeWorker continues polling a pod connectivity status, until the incoming "jobs" channel is closed, and writes results back out to the "results" channel.
 // it only writes pass/fail status to a channel and has no failure side effects, this is by design since we do not want to fail inside a goroutine.
-func probeWorker(k8s *kubeManager, jobs <-chan *ProbeJob, results chan<- *ProbeJobResults, timeoutSeconds int) {
+func probeWorker(prober Prober, jobs <-chan *ProbeJob, results chan<- *ProbeJobResults, timeoutSeconds int) {
 	defer ginkgo.GinkgoRecover()
 	for job := range jobs {
 		podFrom := job.PodFrom
-		connected, command, err := k8s.probeConnectivity(podFrom.Namespace, podFrom.Name, podFrom.Containers[0].Name(), job.PodTo.QualifiedServiceAddress(job.ToPodDNSDomain), job.Protocol, job.ToPort, timeoutSeconds)
+		if job.PodTo.ServiceIP == "" {
```

Comment:
didn't we validate this already?

Reply:
This should never be the case.  but, if it is the case, the probeWorker function should be robust enough to handle it and yield the right error message.  

Maybe in a follow on we should actually just have a `job.Validate()` function ?

Reply:
it is ok, but then it should be 
```suggestion
		if net.ParseIP(job.PodTo.ServiceIP) == nil {
```